pytorch_lightning/callbacks/base.py

```diff
@@ -140,6 +148,18 @@ def on_test_batch_end(
         """Called when the test batch ends."""
         pass
 
+    def on_predict_batch_start(
+        self, trainer, pl_module: LightningModule, batch: Any, batch_idx: int, dataloader_idx: int
+    ) -> None:
+        """Called when the predict batch begins."""
+        pass
+
+    def on_predict_batch_end(
+        self, trainer, pl_module: LightningModule, outputs: Any, batch: Any, batch_idx: int, dataloader_idx: int
+    ) -> None:
+        """Called when the predict batch ends."""
+        pass
+
```

Comment:
could we split this up into a few PRs? The callback hooks could be separate PRs for instance

Reply:
I will do when everything works fine.

Reply:
Opened #7141 for it